### DISCUSSION TITLE
Help Needed: Unable to Identify Dropdown Elements in Appium Inspector

### DISCUSSION THREAD

**Discussion by sait-yamdu** [appium/appium#20831] [Q&A]
Hi everyone,

I’m encountering an issue while testing with Appium Inspector and would appreciate any guidance.

**Issue Details:**
I have an element that, when clicked, displays a dropdown box with additional elements.
However, when I attempt to select or inspect these elements, a large box (blue or yellow in color) appears that covers the entire screen. This makes it impossible to identify the individual elements inside the dropdown.

### **Technical Setup:**

- Appium Version: 2.11.1
- Java Version: The project is configured to use Java 17.
- Automation Framework: UIAutomator2.
- xcuitest@5.15.1
- uiautomator2@3.7.0
- chromium@1.3.28

I’ve tried looking into possible settings or workarounds but haven’t had any success so far.

**Questions:**

1. Has anyone faced a similar issue or knows how to fix this?
2. Are there specific steps or configurations in Appium Inspector to properly identify dropdown elements in this scenario?

By the way changing Native app mode to the Web app mode is not affecting anything. Thank you in advance for your help!

Thank you in advance for your help!

https://github.com/user-attachments/assets/1d900f6b-74cd-443e-91d6-e6d556385791

**Comment by portyu9** [comment]
Hi @sait-yamdu 

First open the dropdown and click **Refresh Source** in Appium Inspector. The Inspector XML is a snapshot and must be refreshed after the UI changes.

If the dropdown items still do not appear in the XML and only a full-screen parent/overlay is shown, then this is not an Inspector selector issue—the application is not exposing those individual items to Android's accessibility/UIAutomator hierarchy.

The fix is to expose each dropdown option as a real accessible element with a stable identifier, for example a `resource-id` or `content-desc`.

Changing from `NATIVE_APP` to `WEBVIEW` will only help if the dropdown is actually inside a WebView.

So the fix is: **refresh the hierarchy after opening the dropdown; if the children are still absent, update the app so the dropdown items are exposed as accessibility nodes.**

Please mark this answer as accepted if it helped, thank you.
